storage/src/test.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::*;
+use storage_core::Data;
+
+decl_schema! {
+    Schema {
+        Map1: Map<Data, Data>,
+    }
+}
+
+#[test]
+fn empty_ro() {
+    utils::concurrency::model(|| {
+        let store = Storage::<_, Schema>::new(inmemory::InMemory::new()).unwrap();
+        let tx = store.transaction_ro();
+        assert_eq!(tx.get::<Map1, _>().get(&b"foo".to_vec()), Ok(None));
+    });
+}
+
+#[test]
+fn empty_rw() {
+    utils::concurrency::model(|| {
+        let store = Storage::<_, Schema>::new(inmemory::InMemory::new()).unwrap();
+        let tx = store.transaction_rw();
+        assert_eq!(tx.get::<Map1, _>().get(&b"foo".to_vec()), Ok(None));
+    });
+}
+
+decl_schema! {
+    // Schema with a compound key
+    Compound {
+        Map2: Map<(String, u16), u64>,
+    }
+}
+
+#[test]
+fn prefix_iteration() {
```

Comment:
Is there a reason this is not part of the test-suite?

Reply:
The test suite only depends on the backend API. Tests that use the higher-level frontend API are here.